src/cmds/hardware/pin/pin.c

```diff
@@ -41,7 +41,18 @@ static int gpio_by_name(char *name) {
 	case '3': case 'd': case 'D': return GPIO_PORT_D;
 	case '4': case 'e': case 'E': return GPIO_PORT_E;
 	case '5': case 'f': case 'F': return GPIO_PORT_F;
+	case '6': case 'g': case 'G': return GPIO_PORT_G;
+	case '7': case 'h': case 'H': return GPIO_PORT_H;
+	case '8': case 'i': case 'I': return GPIO_PORT_I;
+	case '9': case 'j': case 'J': return GPIO_PORT_J;
+	case 'k': case 'K': return GPIO_PORT_K;
 	default:
+		if (strlen(name) > 1 && name[0] == '1') {
```

Comment:
`name = "101"` will return `GPIO_PORT_K`

May be rewrite this function to something like

```
if (toupper(name[0]) >= 'A' && toupper(name[0]) <= 'Z')) {
   idx = toupper(name[0]) - 'A';
} else {
   if (1 != sscanf(name, "%d", &id)) {
        fprintf(stderr, "Failed...");
        return 0;
   }

   return id;
}


```

Reply:
yeah, good suggestion :)

Reply:
Please, take a look.

Reply:
Hm, with a proposed solution

```
pin gpioabcdef 1 toggle
```

will also work.

We should check that the sting is one in size before checking for `a - z`.

Reply:
Fixed.